crates/ruff_workspace/src/configuration.rs

```diff
@@ -780,6 +791,8 @@ impl LintConfiguration {
                 {
                     carryover_ignores = Some(&selection.ignore);
                 }
+
+                docstring_overrides = docstring_override_updates;
```

Comment:
I think it _could_ be fine to omit `docstring_override_updates` and just add to `docstring_overrides` whenever we see a direct selector. In other words, I don't know if this "reset" behavior is important for the purpose of these overrides. I'm trying to think of a case where this would matter.

As an example of why this exists (to the best of my memory): imagine you have `select=D` and `ignore=D401` in your configuration file. If you then pass `--select D4`, we want to ignore the `ignore`. So we treat a `--select` as resetting the selection chain.

Eh, thinking about it further, what you have here is probably right. Imagine you have `select=D415` in your configuration file, and then you run with `--select=D`. We probably still want to enforce the configuration override for `D415` in that case?

Reply:
Hm... I was actually thinking about this in terms of having multiple `ruff.toml`s (e.g. in a monorepo or something). In that case, I think the intended behavior is that `select` "clears" the state of the "parent" `ruff.toml`s, and personally I think it'd be confusing if the docstring conventions were the only counter-example to that.

Reply:
Makes sense.